team and ally damage limiters added to weapon_area_damage_gadget

Gambit99 opened the pull request:
This adds multiplier attributes to the weapon_area_damage gadget. Both default to 1.0. 

Area damage always does zero damage to origin unit. The reasoning is that there is already a nonselfdamage attribute for the unit's weapondef.

Also configured a 0.5 ally damage multipler to arm sonic tanks and tll riot tanks. I play tested this lightly, a large clash with peewees will leave the tanks quite bruised, but mostly intact. 

This type of damage is more effective on offense (hitting stationary or peripheral units) than defense (hitting approaching or swarming units). This is because area damage doesn't push back and stun like, say, the levelers do.

nixtux commented:
[f=0000000] Error: Failed to load: weapon_area_damage.lua  ([string "LuaRules/Gadgets/weapon_area_damage.lua"]:66: 'then' expected near 'if')

Gambit yout changes have some errors, not sure if your aware

nixtux commented:
Got it running with no errors, not checked if it works yet https://gist.github.com/28a7d8e258cdec035538

Gambit99 commented:
Ok, I fixed it -- I did playtest the first time, and it seemed to work fine on my side.

I found the problem and I confess that I am not used to looking for errors in the proper place. I have added a fix commit.
